parsely/src/test/java/com/parsely/parselyandroid/ParselyAPIConnectionTest.kt

```diff
@@ -0,0 +1,129 @@
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.LooperMode
+import org.robolectric.shadows.ShadowLooper.shadowMainLooper
+
+@RunWith(RobolectricTestRunner::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class ParselyAPIConnectionTest {
+
+    private lateinit var sut: ParselyAPIConnection
+    private val mockServer = MockWebServer()
+    private val url = mockServer.url("").toString()
+    private val tracker = FakeTracker()
+
+    @Before
+    fun setUp() {
+        sut = ParselyAPIConnection(tracker)
+    }
+
+    @Test
+    fun `given successful response, when making connection without any events, then make GET request`() {
+        // given
+        mockServer.enqueue(MockResponse().setResponseCode(200))
+
+        // when
+        sut.execute(url).get()
+        shadowMainLooper().idle();
+
+        // then
+        val request = mockServer.takeRequest()
+        assertThat(request).satisfies({
+            assertThat(it.method).isEqualTo("GET")
+            assertThat(it.failure).isNull()
+        })
+    }
+
+    @Test
+    fun `given successful response, when making connection with events, then make POST request with JSON Content-Type header`() {
+        // given
+        mockServer.enqueue(MockResponse().setResponseCode(200))
+
+        // when
+        sut.execute(url, pixelPayload).get()
+        shadowMainLooper().idle();
+
+        // then
+        assertThat(mockServer.takeRequest()).satisfies({
+            assertThat(it.method).isEqualTo("POST")
+            assertThat(it.headers["Content-Type"]).isEqualTo("application/json")
+            assertThat(it.body.readUtf8()).isEqualTo(pixelPayload)
+        })
+    }
+
+    @Test
+    fun `given successful response, when request is made, then purge events queue and stop flush timer`() {
+        // given
+        mockServer.enqueue(MockResponse().setResponseCode(200))
+        tracker.events.add(mapOf("idsite" to "example.com"))
+
+        // when
+        sut.execute(url).get()
+        shadowMainLooper().idle();
+
+        // then
+        assertThat(tracker.events).isEmpty()
+        assertThat(tracker.flushTimerStopped).isTrue
+    }
+
+    @Test
+    fun `given unsuccessful response, when request is made, then do not purge events queue and do not stop flush timer`() {
+        // given
+        mockServer.enqueue(MockResponse().setResponseCode(400))
+        val sampleEvents = mapOf("idsite" to "example.com")
+        tracker.events.add(sampleEvents)
+
+        // when
+        sut.execute(url).get()
+        shadowMainLooper().idle();
+
+        // then
+        assertThat(tracker.events).containsExactly(sampleEvents)
+        assertThat(tracker.flushTimerStopped).isFalse
+    }
+
+    @After
+    fun tearDown() {
+        mockServer.shutdown()
+    }
+
+    companion object {
+        val pixelPayload = """
```

Comment:
The auto-format doesn't work here :/ I didn't formatter it by hand because it looks ok (readable) and I'm concerned about future edits of this field (and a need to format again by hand).